lib/sisimai/bite/json/amazonses.rb

```diff
@@ -0,0 +1,255 @@
+module Sisimai::Bite::JSON
+  # Sisimai::Bite::JSON::AmazonSES parses a bounce object(JSON) which created
+  # by Amazon Simple Email Service.
+  # Methods in the module are called from only Sisimai::Message.
+  module AmazonSES
+    # Imported from p5-Sisimail/lib/Sisimai/Bite/JSON/AmazonSES.pm
+    class << self
+      require 'sisimai/bite/json'
+
+      # https://docs.aws.amazon.com/en_us/ses/latest/DeveloperGuide/notification-contents.html
+      BounceType = {
+        :Permanent => {
+          :General    => '',
+          :NoEmail    => '',
+          :Suppressed => '',
+        },
+        :Transient => {
+          :General            => '',
+          :MailboxFull        => 'mailboxfull',
+          :MessageTooLarge    => 'mesgtoobig',
+          :ContentRejected    => '',
+          :AttachmentRejected => '',
+        },
+      }
+
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def description; return 'Amazon SES(JSON): http://aws.amazon.com/ses/'; end
+
+      # Parse bounce messages from Amazon SES(JSON)
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      # @since v4.20.0
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+        return nil unless mhead['x-amz-sns-message-id']
+        return nil unless mhead['x-amz-sns-message-id'].size > 0
+
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [35.24/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->